rpc/src/rpc.rs

```diff
@@ -2264,16 +2264,25 @@ fn get_spl_token_owner_filter(program_id: &Pubkey, filters: &[RpcFilterType]) ->
         return None;
     }
     let mut data_size_filter: Option<u64> = None;
+    let mut memcmp_filter: Option<&[u8]> = None;
     let mut owner_key: Option<Pubkey> = None;
     let mut incorrect_owner_len: Option<usize> = None;
+    let account_packed_len = TokenAccount::get_packed_len();
     for filter in filters {
         match filter {
             RpcFilterType::DataSize(size) => data_size_filter = Some(*size),
             RpcFilterType::Memcmp(Memcmp {
-                offset: SPL_TOKEN_ACCOUNT_OWNER_OFFSET,
+                offset,
                 bytes: MemcmpEncodedBytes::Bytes(bytes),
                 ..
-            }) => {
+            }) if *offset == account_packed_len && *program_id == inline_spl_token_2022::id() => {
```

Comment:
What do you think about steering token-2022 users toward adding `RpcFilterType::TokenAccountState` instead of a token-2022-specific memcmp?

Then it would be something like:
```
let mut token_account_state_filter: bool = false;
...
match filter {
    RpcFilterType::TokenAccountState => {
        token_account_state_filter = true;
    }
    ...
}
if data_size_filter == Some(account_packed_len as u64) || token_account_state_filter {
    ...
}
```

Reply:
Yeah that makes way more sense.  If someone does search on just the 165th byte though and it's possible to use an index, is it worth supporting that? Or should it use the normal getProgramAccounts implementation?

Reply:
Yeah, I suppose there's no harm in supporting that filter for secondary indexes, and it does provide a way to use the secondary indexes to get only `spl-token-2022` accounts, which could be useful.
So to be explicit regarding this...

> if someone specifies one of those, should we assume that they actually want all token accounts or truly respect what they say?

I'm inclined to respect what they say, which is how your current code will work.

Reply:
My apologies, I hadn't read the whole thing and just assumed that it was simplifying the filter and not adding more filters.  I added the token account state filter as you initially asked.